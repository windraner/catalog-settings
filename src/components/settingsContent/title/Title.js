import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './title.css';

export default class Title extends Component {
  render() {
    const { title, requred } = this.props;

    return (
      <div className="catalog-settings-content__input-title">
        {title}<span className="catalog-settings-content__input-title-star">{requred ? '*' : ''}</span>
      </div>
    );
  }
}

Title.propTypes = {
  title: PropTypes.string,
  requred: PropTypes.bool
};
