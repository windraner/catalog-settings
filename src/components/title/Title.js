import React, { Component } from 'react';

import './title.css';

export default class Title extends Component {
  render() {
    return (
      <div className="catalog-settings__title margin-top-65">
        General settings
      </div>
    );
  }
}
