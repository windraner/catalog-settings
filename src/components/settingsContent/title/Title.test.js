import React from 'react';
import Title from './Title';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

describe('<Title />', () => {
  it('Check text when requred={true}', () => {
    const wrapper = shallow(<Title title="title" requred={true} />);
    expect(wrapper.find('.catalog-settings-content__input-title').length).toBe(1);
    expect(wrapper.find('.catalog-settings-content__input-title').text()).toBe('title*');
  });
  it('Check text when requred={true}', () => {
    const wrapper = shallow(<Title title="title" requred={false} />);
    expect(wrapper.find('.catalog-settings-content__input-title').length).toBe(1);
    expect(wrapper.find('.catalog-settings-content__input-title').text()).toBe('title');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<Title />)
    expect(tree).toMatchSnapshot()
  })
});
