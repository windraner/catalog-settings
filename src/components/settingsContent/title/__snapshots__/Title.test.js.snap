// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <span
          className="catalog-settings-content__input-title-star"
        >
          
        </span>,
      ],
      "className": "catalog-settings-content__input-title",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "catalog-settings-content__input-title-star",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <span
            className="catalog-settings-content__input-title-star"
          >
            
          </span>,
        ],
        "className": "catalog-settings-content__input-title",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "catalog-settings-content__input-title-star",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
